packages/offix-datastore/src/predicates/PredicateFunctions.ts

```diff
@@ -0,0 +1,62 @@
+export abstract class PredicateFunction {
+    public filter(models: any[]) {
+        return models.filter((m) => this.evaluate(m));
+    }
+
+    public abstract evaluate(model: any): boolean;
+}
+
+export class ModelFieldPredicate extends PredicateFunction {
+    constructor(
+        private key: string,
+        private value: string,
+        private operator: Function
+    ) {
+        super();
+    }
+
+    public evaluate(model: any) {
+        return this.operator(model[this.key], this.value);
+    }
+}
+
+export interface ExpressionOperator {
+    operate(previousResult: boolean, currentResult: boolean): boolean;
+}
+
+export const ExpressionOperators = {
```

Comment:
Total rocket science! I'm not longer impressed this is so clever that I cannot comprehend it :D

Reply:
Thank you! I just avoided switch statements. Doing it this way all us to obey the Open-Closed principle. Now, you can add an `ExpressionOperator` without modifying the `PredicateExpression` class